src/main/java/net/revelc/code/impsort/maven/plugin/AbstractImpSortMojo.java

```diff
@@ -239,14 +279,27 @@ public final void execute() throws MojoExecutionException, MojoFailureException
         getLog().debug("Reading file " + path);
 
         try {
-          Result result = impSort.parseFile(path);
-          result.getImports().forEach(imp -> getLog().debug("Found import: " + imp));
-          if (result.isSorted()) {
+          byte[] buf = Files.readAllBytes(path);
+          String newHash = getHash(buf);
+          String key = baseDir.relativize(path).toString();
+          String prvHash = hashCache.getProperty(key);
+          if (prvHash != null && prvHash.equals(newHash)) {
             numAlreadySorted.getAndIncrement();
+            getLog().debug("Unchanged: " + path);
           } else {
-            numProcessed.getAndIncrement();
+            Result result = impSort.parseFile(path, buf);
+            result.getImports().forEach(imp -> getLog().debug("Found import: " + imp));
+            if (result.isSorted()) {
+              numAlreadySorted.getAndIncrement();
+            } else {
+              numProcessed.getAndIncrement();
+            }
+            processResult(path, result);
+            buf = Files.readAllBytes(path);
```

Comment:
Should `processResult` return the new file contents, so that way we don't have to read the file a second time?

Reply:
Yes, I didn't want to be too disruptive, but this is clearly not optimal.  I'll work on a follow up commit to fix that.